src/repoutil.js

```diff
@@ -0,0 +1,82 @@
+const core = require('@actions/core')
+// probably dont need le octokit, this is githubs rest client 
+const { Octokit } = require('@octokit/rest')
+
+
+module.exports.isTruthy = function isTruthy (val) {
+    if (val.toString().toLowerCase() === 'true') {
+        return true
+    } else {
+        return false
+    }
+}
+
+module.exports.isValidType = function isValidType (val) {
+    let formattedVal = val.toString().toLowerCase()
+    
+    if (formattedVal === 'docker' || formattedVal === "container" ) {
+        return formattedVal
+    } else {
+        throw new Error("Invalid package-type! Must be type <docker> or <container>") 
+    }
+}
+
+
+
+// create a function that deletes everything older than the last 600
+// use the constant numKeep
+// if the total amount of packages is < 600 exit with nice message
+// le yeet
+
+async function createOcto(cfg){
+    return new Octokit({
+        auth: cfg.ghToken
+    })
+}
+
+module.exports.getPackages = async function getPackages(cfg){
+    let octoClient = await createOcto(cfg)
+
+    let pkg = await octoClient.paginate( await octoClient.rest.packages.getAllPackageVersionsForPackageOwnedByOrg({
+        package_type: cfg.packageType,
+        package_name: cfg.packageName,
+        org: cfg.repoOwner,
+        state: "active",
+        per_page: 50
+    }))
+
+
+// pkgs are returned sorted by most recently created
+return pkg
+}
+
+module.exports.parsePackages = async function parsePackages(cfg,pkg){
+console.log("Were keepin " + cfg.numKeep)
+    do {
+        let deletePkg = pkg.pop()
+        // console.log("Here comes deletePkg")
+        // console.log(deletePkg)
+        await this.deletePackages(deletePkg,cfg)
+    } while (pkg.length >= cfg.numKeep);
+
+    return pkg
+}
+
+module.exports.deletePackages = async function deletePackages (pkg,cfg){
+    let octoClient = await createOcto(cfg)
+
+    console.log("deletePkg Function")
+    console.log(pkg)
+    console.log("package_type = " + pkg.metadata.package_type)
+    console.log("package_name = " + pkg.name)
+    console.log("org = " + cfg.repoOwner)
+    console.log('end')
+    let result = await octoClient.rest.packages.deletePackageForOrg({
```

Comment:
@ryanjfrizzell For some reason it cannot find the package no matter what I've tried here....  I must be missing something.

Reply:
Output:
```
deletePkg Function
{
  id: 3844109,
  name: 'sha256:fdf47fdd9e2f388f1570c6c13b41a02c8bba960fcccbe8539842d076537046c6',
  url: 'https://api.github.com/orgs/CirrusMD/packages/container/github-action-clean-docker-packages-test/versions/3844109',
  package_html_url: 'https://github.com/orgs/CirrusMD/packages/container/package/github-action-clean-docker-packages-test',
  created_at: '2021-07-13T17:38:47Z',
  updated_at: '2021-07-13T17:38:47Z',
  html_url: 'https://github.com/orgs/CirrusMD/packages/container/github-action-clean-docker-packages-test/3844109',
  metadata: { package_type: 'container', container: { tags: [Array] } }
}
package_type = container
package_name = sha256:fdf47fdd9e2f388f1570c6c13b41a02c8bba960fcccbe8539842d076537046c6
org = CirrusMD
end
(node:4991) UnhandledPromiseRejectionWarning: HttpError: Package not found.
```